### PR TITLE
Select runner image based on charmcraft.yaml build-on

### PR DESCRIPTION
**Pull request by arturo-seijas**
Select the base in the charmcraft.yaml file for build as runner image for the different jobs.

### CODE REVIEW

**Review by gtrkiller — DISMISSED** [review]
LGTM

**Review by gtrkiller — DISMISSED** [review]

**Review by gtrkiller — DISMISSED** [review]
LGTM

File: .github/workflows/get_runner_image.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+jobs:
+  get-runner-image:
+    name: Get runner image
+    runs-on: ubuntu-22.04
+    outputs:
+      runs-on: ${{ env.BUILD_ON }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Get build-on and run-on values
```
Comment (mthaddon) [inline .github/workflows/get_runner_image.yaml:18]: We're just getting `build-on` here, so should we update the description to reflect that?

File: .github/workflows/get_runner_image.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        run: |
+          echo "BUILD_ON=$(yq '[.bases.[].build-on.[].name, .bases.[].build-on.[].channel] | join("-")' charmcraft.yaml)" >> $GITHUB_ENV
```
Comment (mthaddon) [inline .github/workflows/get_runner_image.yaml:20]: You can specify multiple `build-on` targets in `charmcraft.yaml`, so I think we should adjust this to:

```yq '[.bases.[].build-on.[0].name, .bases.[].build-on.[0].channel] | join("-")' charmcraft.yaml```

**Review by mthaddon — DISMISSED** [review]
LGTM, thx

**Review by gtrkiller — APPROVED** [review]
LGTM

**Review by mthaddon — APPROVED** [review]
LGTM

**Pull request merged**